modules/sdk-coin-sui/test/unit/sui.ts

```diff
@@ -56,13 +56,64 @@ describe('SUI:', function () {
   });
 
   describe('Verify transaction: ', () => {
-    it('should succeed to verify transaction in base64 encoding', async function () {
+    it('should succeed to verify transaction', async function () {
       const txPrebuild = newTxPrebuild();
       const txParams = newTxParams();
       const verification = {};
       const isTransactionVerified = await basecoin.verifyTransaction({ txParams, txPrebuild, verification });
       isTransactionVerified.should.equal(true);
     });
+
+    it('should fail to verify transaction with invalid param', async function () {
+      const txPrebuild = {};
+      const txParams = newTxParams();
+      txParams.recipients = undefined;
+      await basecoin
+        .verifyTransaction({
+          txParams,
+          txPrebuild,
+        })
+        .should.rejectedWith('missing required tx prebuild property txHex');
+    });
+  });
+
+  describe('Explain Transaction: ', () => {
+    it('should explain a transfer transaction', async function () {
+      const explainedTransaction = await basecoin.explainTransaction({
+        txHex: testData.TRANSFER_TX,
+      });
+      explainedTransaction.should.deepEqual({
+        displayOrder: ['id', 'outputs', 'outputAmount', 'changeOutputs', 'changeAmount', 'fee', 'type'],
```

Comment:
what does outputAmount and changeOutputs refer to here? Does SUI have these properties in Object model?

Reply:
If SUI doesn't have concept of change, I'd say remove those properties, and make SUI's own native tx explainTransaction() as opposed to using baseTransaction.explainTransaction()

Reply:
SUI doesn't have these properties, but these fields are not optional under `interface ITransactionExplanation`

Reply:
Making SUI's own SuiTransactionExplanation also works and makes sense, but I checked all other coins and they return these properties even if they don't have them like SUI. So I'm wondering if I should follow the current behaviour or just make it Sui specific.

Reply:
Interesting...weird that all other coins would have these too, I'd say it's probably fine then. Something to maybe revisit as a backlog item for altcoins team later on :)

Reply:
agree, it'd be something to revisit and clean up later!